bndtools.builder/src/org/bndtools/builder/BuilderPlugin.java

```diff
@@ -1,12 +1,24 @@
 package org.bndtools.builder;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.ui.IDecoratorManager;
+import org.eclipse.ui.plugin.AbstractUIPlugin;
 import org.osgi.framework.BundleContext;
 
-public class BuilderPlugin extends org.eclipse.core.runtime.Plugin {
+public class BuilderPlugin extends AbstractUIPlugin {
```

Comment:
Try this instead
```
PlatformUI.getWorkbench().getDecoratorManager()
```

Then we don't have to change the class hierarchy of this to AbstractUIPlugin.